thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -155,10 +155,11 @@ class CDistributorBase : implements IHashDistributor, implements IExceptionHandl
                     }
                     else
                     {
-                        prev.set(row);
-                        rows.enqueue(row.getClear());
+                        rows.enqueue(prev.getClear());
+                        prev.setown(row.getClear());
```

Comment:
It's not a legacy bug, it was part of best changes.
For reference, this was from our IM conv. on Friday when I spotted the crash:
```
Jake
have a test that fails (crashes)
Nov 25, 2016 15:54
Thread #123 [thorslave_mytho] 7694 [core: 4] (Suspended : Signal : SIGSEGV:Segmentation fault)    
    cAc3::CompareClass::docompare(void const*, void const*) const at 0x7ff25bd1ca19    
    CDistributorBase::CSendBucket::dedup() at thhashdistribslave.cpp:150 0x7ff85882afbf    
    CDistributorBase::CSender::dedup() at thhashdistribslave.cpp:541 0x7ff85882c66f    
    CDistributorBase::CSender::CWriteHandler::main() at thhashdistribslave.cpp:426 0x7ff85882bf2a    
    CPooledThreadWrapper::run() at jthread.cpp:855 0x7ff856b77a08    
    Thread::begin() at jthread.cpp:272 0x7ff856b74e1b    
    Thread::_threadmain() at jthread.cpp:118 0x7ff856b748a8    
    start_thread() at pthread_create.c:312 0x7ff8562e2182    
    clone() at clone.S:111 0x7ff85600f47d    
crashed prior to my changes
ic it's here:
OwnedConstThorRow prev = dedupList.getClear(--c);
if (!keepBestCompare)
{
    rows.enqueue(prev.getClear());
    for (unsigned i = c; i>0;)
    {
        OwnedConstThorRow row = dedupList.getClear(--i);
        if (0 == iCompare->docompare(prev, row))
        {
            /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
            * which will be deducted when sent.
            */
        }
        else
        {
            prev.set(row);
            rows.enqueue(row.getClear());
        }
    }
}
prev has been cleared prior to docompare
think needs to be:
OwnedConstThorRow prev = dedupList.getClear(--c);
if (!keepBestCompare)
{
    for (unsigned i = c; i>0;)
    {
        OwnedConstThorRow row = dedupList.getClear(--i);
        if (0 == iCompare->docompare(prev, row))
        {
            /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
            * which will be deducted when sent.
            */
        }
        else
        {
            rows.enqueue(prev.getClear());
            prev.set(row);
        }
    }
    rows.enqueue(prev.getClear());
}
```